apps/full-site-editing/full-site-editing-plugin/starter-page-templates/class-starter-page-templates.php

```diff
@@ -217,6 +234,13 @@ public function clear_sideloaded_image_cache( $id ) {
 		}
 	}
 
+	/**
+	 * Deletes cached templates data when theme switches.
+	 */
+	public function clear_templates_cache() {
+		delete_transient( $this->templates_cache_key );
```

Comment:
I like that we're cleaning up after ourselves rather than using theme slug to cache bust. That said, I was wondering why we didn't also use the theme slug in the transient key.